tests/system/test_fim/test_fim_synchronization/test_fim_synchronization.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+
+import os
+
+import pytest
+from wazuh_testing.tools import WAZUH_LOGS_PATH
+from wazuh_testing.tools.monitoring import HostMonitor
+from wazuh_testing.tools.system import HostManager
+
+
+# Hosts
+testinfra_hosts = ["wazuh-manager", "wazuh-agent1"]
+
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                              'provisioning', 'one_manager_agent', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+messages_path = [os.path.join(local_path, 'data/messages.yml'),
+                 os.path.join(local_path, 'data/delete_message.yml'),
+                 os.path.join(local_path, 'data/wait_fim_scan.yml')]
+tmp_path = os.path.join(local_path, 'tmp')
+scheduled_mode = 'testdir1'
+
+
+def create_folder(folder_path):
+    # Create folder
+    host_manager.run_command('wazuh-agent1', f'mkdir {folder_path}')
+
+    # Create file
+    host_manager.run_command('wazuh-agent1', f'touch {folder_path}/{folder_path}.txt')
+
+
+def clean_logs():
+    host_manager.clear_file(host='wazuh-manager', file_path=os.path.join(WAZUH_LOGS_PATH, 'ossec.log'))
+    host_manager.clear_file(host='wazuh-agent1', file_path=os.path.join(WAZUH_LOGS_PATH, 'ossec.log'))
+
+
+def wait_for_fim_scan_end(folder_path):
+    try:
+        HostMonitor(inventory_path=inventory_path,
+                    messages_path=messages_path[2],
+                    tmp_path=tmp_path).run()
+    finally:
+        host_manager.run_command('wazuh-agent1', f'rm -rf {folder_path}')
+
```

Comment:
Could these methods be parameterized and added outside of the test to be used by other tests in the future or would it cause problems?

Reply:
Done in commit [5972a](https://github.com/wazuh/wazuh-qa/pull/2443/commits/5972a0084b9b136aed65f0468b971bfbb477cc2e)